stimuli_ci_generation/src/server/stimuli_ci.py

```diff
@@ -60,14 +60,21 @@ def generate_ci(identifier, file_name):
         download_dir(gcp_config.STIMULI_IMG_BUCKET, identifier, f"{stimuli_dir}")
         print(f"downloaded stimuli images to {ws_dir}/stimuli")
 
+    print(f"stimuli has {len(os.listdir(f'{ws_dir}/stimuli'))} files.")
+    if len(os.listdir(f"{ws_dir}/stimuli")) == 0:
+        raise Exception("CI could not be generated due to empty stimuli directory.")
+
+    if file_name != USER_SELECTION:
+        raise Exception(f"name of user selections file does not match {USER_SELECTION}")
     download_file(
         gcp_config.USER_SELECTION_BUCKET,
-        f"{identifier}/{USER_SELECTION}",
+        f"{identifier}/{file_name}",
         f"{ws_dir}/{USER_SELECTION}",
     )
-    print(f"user_selection.csv has been downloaded to {ws_dir}/{USER_SELECTION}")
+    print(f"user_selection.csv has been downloaded to {ws_dir}/{file_name}")
 
     try:
+        print("started running generate_ci.R")
         subprocess.check_call(["Rscript", r_script_path, ws_dir], shell=False)
         print("Finished running generate_ci.R")
```

Comment:
do you have an estimate of how time this takes on hand? If not, let's add a simple timer here to give us something for presentation then remove later :)

Reply:
not exact but within seconds, we can add a timer for sure

Reply:
ooh that's awesome. Then let's not worry about it